### ISSUE TITLE
Amuleweb stops at random

### ISSUE THREAD

**Issue by Kayn2106** [ngosang/docker-amule#38]
No matter what amuleweb stops randomly with the logs: Command ''amuleweb' '--amule-config-file=/home/amule/.aMule/amule.conf''..... has finished with status code '-6'

This can happened once a day as well as 10 times in an hour, I wasn't able to understand what causes this problem; however it seems a long known problem 'cause I found old threads around the web that mention this behavior, so I'm guessing that's an old problem that has never been fixed. Anyway, let's move on. Since aMule keeps working properly, I'm wondering if it could be possible to write a script that checks if amuleweb is stopped and if needed restarts it or maybe add a modification that does this, just like the kad graph fix and the others...

**Comment by ngosang** [comment]
Since aMule is not maintained this will be a good improvement.
Some notes:
* The entypoint starts `amuled -c /home/amule/.aMule -o`
* `amuled` reads the `amule.conf` file, the section `[WebServer]  Enabled=1` and it starts `amuleweb --amule-config-file=/home/amule/.aMule/amule.conf`
* If the process `amuled` crash it's restarted by the entrypoint
* If the process `amuleweb` crash it's not restarted by `amuled`

Possible fix:
* Set `[WebServer]  Enabled=0` so `amuled` does not start `amuleweb`
* Start `amuleweb` from the entrypoint but we can't use the `amule.conf` file. Inside the Docker container you can run `man amuleweb` to see more details.
* We should read the `amule.conf` file using bash and set the equivalent parameters in `amuleweb`

Improvement:
* Instead of doing a lot of things in the entrypoint, we should use `s6` to keep running the processes. This project has a good example => https://github.com/kanboard/kanboard/tree/main/docker

I don't have time to work on this. Any PR is welcome.

**Comment by puigru** [comment]
Experiencing the exact same issue

**Comment by Kayn2106** [comment]
> Experiencing the exact same issue

Yeah, that's driving me crazy. Unfortunately I don't have the knowledge to solve this so....

**Comment by garret** [comment]
I have been experiencing that that the webui and also access via gui ui is extremely laggish after update to latest docker image version. I reverted to previous version (from around october 2023) and experience is smooth. I advice to try to revert back to previous image and see if you still experience the same issues. For the moment, I will stay with the older version since it just works.

**Comment by Kayn2106** [comment]
> I have been experiencing that that the webui and also access via gui ui is extremely laggish after update to latest docker image version. I reverted to previous version (from around october 2023) and experience is smooth. I advice to try to revert back to previous image and see if you still experience the same issues. For the moment, I will stay with the older version since it just works.

I've started using this container back in the mid 2023 and the issues was already there and unfortunately no changes in the next releases... However right now I don't know why, maybe due to the recent Unraid upgrade to 6.12.6, I think it's more stable I mean I've notice less webui crash...

**Comment by puigru** [comment]
It's a shame to see that ed2k clients / web interfaces are in such disrepair. I used to use mldonkey which seemed to be more stable but was similarly dated and has other issues. It would be nice if someone took the time to make a modern interface using the amule's EC protocol. That way we could have a better ed2k client without requiring a full on reimplementation.

Something like https://github.com/mwarning/p2p-gui

**Comment by Kayn2106** [comment]
I've spoken too soon. It's crashing again like before: four times only in the last hour and I didn't change anything on my NAS...

**Comment by puigru** [comment]
For anyone still running into this, I've switched back to mldonkey. It's been working like a champ and it arguably has a better UI
I'm using this image: https://github.com/Wibol/mldonkey-docker

**Comment by garret** [comment]
I confirm I have been running the previous version for many months without any issues. So my suggestion is to not update to the latest tag version on docker. Have you tried that?

I know very well mldonkey but I much prefer the interface of the personalized skin of amule coming with this image :)

**Comment by ngosang** [comment]
> I confirm I have been running the previous version for many months without any issues. So my suggestion is to not update to the latest tag version on docker. Have you tried that?

Could you try with the latest release? https://github.com/ngosang/docker-amule/releases/tag/2.3.3-18

**Comment by garret** [comment]
@ngosang I just updated to it. I will write a new reply to this issue if I start getting the same issue. Have you just updated the image to the latest alpine package version? Thank you in the meantime.

**Comment by garret** [comment]
Just experienced after 2 days of running the container non stop that the webinterface does not work anymore. So I can only connect via amule client (but everything else work, so the downloading/uploading thing). I just restarted the container again. If it happens again I guess I will come back again to v16.

**Comment by ngosang** [comment]
@garret You can try the "auto restart" and "fix kad stats" mods = https://github.com/ngosang/docker-amule?tab=readme-ov-file#modifications

**Comment by garret** [comment]
I have always had those options enabled since I used your image ;) In any case, I have been continuing using the latest release at the moment of writing. The image has always worked on the background 24/7 and only maybe 3 times in total I had the issue where the web interface went down but amule core was still running. In the meanwhile, I also switched from a raspberry pi server to a nixos mini pc. Coincidentally, today I had an issue where all the 32GB of the ram got used and all the 4 processors of my N100 server. Restarting the container did not help. I had to cancel some files from the amule config to solve. I could see the log and known2.met file were above 180MB. After cancelling those files, everything seems again normal. All in all, I am satisfied with the image and will keep it running.

**Comment by jaitor1** [comment]
I am having same issue in the latest release

**Comment by alejandrodt** [comment]
same here with the latest release :(

**Comment by S0LERA** [comment]
Same issue heer with latest release

**Comment by feanorknd** [comment]
same issue... if I refresh multiple times or stay at the web interface for minutes, web interface won't be accessible again until restarting the container.

**Comment by garret** [comment]
To me now this happens like 4 times per year and I have made my mind that I can live with that. It might also depend on the now rusty code of aMule.

**Comment by ngosang** [comment]
aMule several bugs and the original developers are unreachable.
You can try the "auto restart" and "fix kad stats" mods = https://github.com/ngosang/docker-amule?tab=readme-ov-file#modifications

**Comment by garret** [comment]
@ngosang I already wrote last year in August in this issue that I have always had both your mods enabled and that did not change the result ;)

**Comment by Lordpedal** [comment]
The best solution to web interface problems is to use the EC protocol through port 4712, and for this, the project mentioned by @got3nks at [aMule-Web-Controller](https://github.com/got3nks/amule-web-controller) is the best solution.

**Comment by ngosang** [comment]
I fixed the issue in https://github.com/ngosang/docker-amule/commit/ec7be603709add4b446274a0b7850fc63070e280 https://github.com/ngosang/docker-amule/commit/64320e308aee3bb99ff1f9501fd6f3973f032901
